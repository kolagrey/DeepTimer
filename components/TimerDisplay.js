import React from 'react';
import { View, Text } from 'react-native';
import PropTypes from 'prop-types';
import BlinkingText from './BlinkingText';
import styles from '../styles';

const TimerDisplay = (props) => {
    const { timerDisplay, turnRed, startBlinking } = props;

    const getStyle = (style, isRed = false) => {
        if (isRed) {
            return style.timerTextRed;
        }
        return style.timerTextBlack;
    };

    return (
        <View style={styles.timerTextDisplayArea}>
            {startBlinking ?
                <BlinkingText text={timerDisplay} style={getStyle(styles, turnRed)} /> :
                <Text style={getStyle(styles, turnRed)}>{timerDisplay}</Text>
            }
        </View>
    );
};

TimerDisplay.propTypes = {
    timerDisplay: PropTypes.string,
    turnRed: PropTypes.bool,
    startBlinking: PropTypes.bool
};

export default TimerDisplay;