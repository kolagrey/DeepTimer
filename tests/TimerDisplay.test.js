import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';

import TimerDisplay from '../components/TimerDisplay';

describe('<TimerDisplay/>', () => {

    test('renders correctly', () => {
        const tree = renderer.create(<TimerDisplay/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("is of type 'View'", () => {
        const tree = renderer
            .create(
                <TimerDisplay />,
            ).toJSON();
        expect(tree.type).toBe('View');
    });

    it('can hold a child', () => {
        const tree = renderer
            .create(
                <TimerDisplay>
                    <Text>
                        Hello Test
                    </Text>
                </TimerDisplay>,
            ).toJSON();
        expect(tree.children.length).toBe(1);
    });
});