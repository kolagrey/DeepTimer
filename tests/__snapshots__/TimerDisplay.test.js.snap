// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimerDisplay/> renders correctly 1`] = `
<View
  style={
    Object {
      "margin": 5,
      "width": 250,
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 90,
        "fontWeight": "300",
      }
    }
  />
</View>
`;
